changelog/unreleased/kong/fix-target-deletion.yml

```
@@ -0,0 +1,3 @@
+message: fixed an issue where targets can't be removed from dns query if they are deleted or updated via admin API.
```

One nit, feel free to ignore.

```suggestion
message: Fixed an issue where targets can't be removed from dns query if they were deleted or updated via admin API.
```